src/libjasper/base/jas_icc.c

```diff
@@ -1620,8 +1622,9 @@ static int jas_iccputuint(jas_stream_t *out, unsigned n, jas_ulonglong val)
 
 static int jas_iccputsint(jas_stream_t *out, unsigned n, jas_longlong val)
 {
+	assert(val >= 0);
 	jas_ulonglong tmp;
-	tmp = (val < 0) ? (abort(), 0) : val;
+	tmp = (val < 0) ? 0 : val;
```

Comment:
val < 0 will now never be the case, will it? So I don't think we need `tmp`. Except maybe for different type `jas_longlong` vs `jas_ulonglong`.

Reply:
This commit tries to do least-intrusive refactoring - on old code which didn't make sense. *Everything* about this function is completely wrong! I have no idea what the author had in mind.
But my commit doesn't even try to fix anything about that function - it only refactors an `abort()` which also affects release builds to `assert()` (debug builds only), but keeps the negative cropping (which never worked).

Reply:
I see. GTM then.

Reply:
> val < 0 will now never be the case, will it? So I don't think we need `tmp`. Except maybe for different type `jas_longlong` vs `jas_ulonglong`.

@MaxKellermann My best guess of what happened here is that there wasn't a function to handle reading signed long long values and I knew that negative values were not needed at the time.  So, I used the already existing function for unsigned data output and aborted if the value to be output was negative.  The intention was probably to fix this at a later point, and this never got fixed.  Is this theory consistent with what you see?  Again, it is my best guess.  And if this explanation is correct, yes, it is gross, but... Please keep in mind that JasPer was originally written by me alone as a side project while I was doing my Ph.D..  JasPer was not directly related to my Ph.D. and in many ways was very detrimental to my progress on my Ph.D..  So, I was under tremendous time pressure to get things working in JasPer, no matter how gross some implementation details might be. Some of the code that I wrote back then makes me cringe when I look at today, but the alternative (if I didn't cut some corners) would have been that I spent so much time on JasPer that I would never have finished by Ph.D..